lib/mcp/mcp-server.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @fileoverview MCP Server for handling requests and responses to ESLint.
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const { McpServer } = require("@modelcontextprotocol/sdk/server/mcp.js");
+const { z } = require("zod");
+const { ESLint } = require("../eslint");
+const pkg = require("../../package.json");
+
+//-----------------------------------------------------------------------------
+// Server
+//-----------------------------------------------------------------------------
+
+const mcpServer = new McpServer({
+	name: "ESLint",
+	version: pkg.version,
+});
+
+const filePathsSchema = {
+	filePaths: z.array(z.string()).nonempty(),
```

Comment:
try to be a lot more explicit here, otherwise the superagent/orchestrator/mcp client might give you the wrong path, something like this:

```suggestion
	filePaths: z.array(z.string().min(1).describe('Full filesystem path a file to be analyzed')).nonempty(),
```

I suspect some evolution is going to be needed around this at the MCP level.